Runtime/Scripts/PersistentConnection.cs

```diff
@@ -131,192 +217,193 @@ IEnumerator OnIntervalEdgeEventsLocation(EdgeEventsConnection connection)
-              }
-              break;
-            case EventLatencyProcessed:
-              Logger.Log("Received EventLatencyProcessed, Comparing Latency to the LatencyThreshold from configs.");
-              if (fcTriggers.Contains(FindCloudletEventTrigger.LatencyTooHigh))
-                {
-                  if (config.latencyThresholdTriggerMs.CompareTo(edgeEvent.Statistics.Avg) <= 0)
-                  {
-                    integration.UseFindCloudletPerformanceMode(true);
-                    bool fcResult = await integration.FindCloudlet();
-                    integration.UseFindCloudletPerformanceMode(false);
-                    if (fcResult == true)// fixme check if automigration is true
-                    {
-                      findCloudletEvent.trigger = FindCloudletEventTrigger.LatencyTooHigh;
-                      findCloudletEvent.newCloudlet = integration.FindCloudletReply;
-                      integration.NewFindCloudletHandler(EdgeEventsStatus.success, findCloudletEvent);
-                    }
-                  }
-                }
-              break;
-            case EventCloudletUpdate:
-              Logger.Log("Received Cloudlet Update, New CloserCloudlet found,New Cloudlet GPS :" + CloudletLocation.GpsLocationFieldNumber);
-              if (fcTriggers.Contains(FindCloudletEventTrigger.CloserCloudlet))
-              {
-                findCloudletEvent.trigger = FindCloudletEventTrigger.CloserCloudlet;
-                findCloudletEvent.newCloudlet = edgeEvent.NewCloudlet;
-                bool fcResult = await integration.FindCloudlet();
-                integration.NewFindCloudletHandler(EdgeEventsStatus.success, findCloudletEvent);
-              }
-              break;
-            case EventError:
-              Logger.Log("Received EventError, :" + edgeEvent.ErrorMsg);
-              FindCloudletEvent fcEvent = new FindCloudletEvent() { trigger = FindCloudletEventTrigger.Error };
-              integration.NewFindCloudletHandler(EdgeEventsStatus.error, fcEvent);
-              break;
-            default:
-            case EventUnknown:
-              Logger.Log("Received Unknown Event");
-              break;
+            PropagateError(FindCloudletEventTrigger.LatencyTooHigh,
+              "New Cloudlet obtained from FindCloudletPerformanceMode is the same as old cloudlet");
           }
         }
+        else
+        {
+          PropagateError(FindCloudletEventTrigger.LatencyTooHigh, "FindCloudletPerformanceMode failed");
+        }
+      }
+    }
 
-    private void OnApplicationFocus(bool focus)
+    async void CompareLatencies(Statistics receivedStats)
     {
-      if (locationUpdatesRunning)
+      AppPort appPort;
+      Site site;
+      var netTest = new NetTest(integration.matchingEngine);
+      if (hasTCPPorts)
+      {
+        appPort = integration.GetAppPort(LProto.Tcp, config.latencyTestPort == 0 ? 0 : config.latencyTestPort);
```

Comment:
same comment as above about ternary operator

Reply:
this should be an easy fix

Reply:
Will need to fix that.